File: ShoppingAppApi/ShoppingAppApi/Models/Order.cs
namespace ShoppingAppApi.Models;

public class Order
{
    public Guid Id { get; init; }
    public DateTime CreatedAtUtc { get; init; } = DateTime.UtcNow;
    public required List<OrderItem> Items { get; init; }
    public required bool PaymentSucceeded { get; init; }
    public Guid IdempotencyToken { get; init; }
}

File: ShoppingAppNg/src/app/models/order.ts
export interface CreateOrderItemRequest {
  productId: string;
  quantity: number;
}

export interface CreateOrderRequest {
  items: CreateOrderItemRequest[];
  forcePaymentFailure: boolean;
  idempotencyToken: string;
}

export interface OrderItem {
  productId: string;
  quantity: number;
  unitPrice: number;
}

export interface Order {
  id: string;
  createdAtUtc: string;
  items: OrderItem[];
  paymentSucceeded: boolean;
  idempotencyToken: string;
}
